src/required_resources.cpp

```diff
@@ -142,6 +133,41 @@ int RequiredResources::create_maps_impl(BPFtrace &bpftrace, bool fake)
   return failed_maps;
 }
 
+template <typename T>
+std::unique_ptr<T> RequiredResources::prepareFormatStringDataMap(
+    std::vector<std::tuple<FormatString, std::vector<Field>>> args,
+    bool fake)
```

Comment:
I think it'd be nicer to take the `is_invalid_map` lambda as an argument instead of repeating the logic using the `fake` bool